mtags/src/main/scala-2/scala/meta/internal/pc/PcCollector.scala

```diff
@@ -491,4 +491,13 @@ abstract class PcCollector[T](
     }
   }
 
+  // We don't want to collect synthethic `map`, `withFilter`, `foreach` and `flatMap` in for-comprenhensions
+  private def isForComprehensionMethod(sel: Select): Boolean = {
+    val name = sel.name.decoded
+    val syntheticName =
+      name == "map" || name == "withFilter" || name == "flatMap" || name == "foreach"
+    val wrongSpan = sel.qualifier.pos.includes(sel.namePosition.focusStart)
+    syntheticName && wrongSpan
+  }
+
```

Comment:
```suggestion
  private val forCompMethods = Set(nme.map, nme.flatMap, nme.withFilter, nme.foreach)
  
  // We don't want to collect synthethic `map`, `withFilter`, `foreach` and `flatMap` in for-comprenhensions
  private def isForComprehensionMethod(sel: Select): Boolean = {
    val syntheticName = sel.name match {
      case name: TermName =>  forCompMethods(name)
      case _ => false
    }
    val wrongSpan = sel.qualifier.pos.includes(sel.namePosition.focusStart)
    syntheticName && wrongSpan
  }
```
and something along these lines for Scala 3?